code/modules/mob/mob_say.dm

```diff
@@ -100,22 +100,47 @@
 	return LINGHIVE_NONE
 
 /**
-  * Get the mode of a message
+  * Extracts and cleans message of any extenstions at the begining of the message
+  * Inserts the info into the passed list, returns the cleaned message
   *
   * Result can be
+  * * SAY_MODE (Things like aliens, channels that aren't channels)
   * * MODE_WHISPER (Quiet speech)
   * * MODE_SING (Singing)
   * * MODE_HEADSET (Common radio channel)
-  * * A department radio (lots of values here)
+  * * RADIO_EXTENSION the extension we're using (lots of values here)
+  * * RADIO_KEY the radio key we're using, to make some things easier later (lots of values here)
+  * * LANGUAGE_EXTENSION the language we're trying to use (lots of values here)
   */
-/mob/proc/get_message_mode(message)
-	var/key = message[1]
-	if(key == "#")
-		return MODE_WHISPER
-	else if(key == "%")
-		return MODE_SING
-	else if(key == ";")
-		return MODE_HEADSET
-	else if((length(message) > (length(key) + 1)) && (key in GLOB.department_radio_prefixes))
-		var/key_symbol = lowertext(message[length(key) + 1])
-		return GLOB.department_radio_keys[key_symbol]
+#define MESSAGE_MODS_LENGTH 6 //The amount of items we are looking for in the message

```

Comment:
Don't put the define between the proc definition and the documentation comment, I can't imagine it'll document properly.

Also document the MESSAGE_MODS_LENGTH define with a dmdoc compliant comment (```/// The amount of items we are looking for in the message`` above the define, not beside it)

Reply:
Didn't know that worked, thank you bobbah